AdversarialSearch/Minimax/TicTacToeMinimax.h

```diff
@@ -2,29 +2,22 @@
 #define _TIC_TAC_TOE_MINIMAX_H
 
 #include <vector>
-#include <cmath>
-#include <iostream>
-#include <cstdio>
 
-extern unsigned int Playing;
-extern unsigned int Draw;
-extern unsigned int CrossWins;
-extern unsigned int CircleWins;
-
-class TicTacToeMinimax
+namespace TicTacToeMinimax
 {
-    friend class MinimaxBenchmarker;
-    private:
-        float getMaxTreeScore(unsigned int board, float alpha, float beta, int depth);
-        float getMinTreeScore(unsigned int board, float alpha, float beta, int depth);
-        unsigned int evaluateMaxDecision(unsigned int board);
-        unsigned int evaluateMinDecision(unsigned int board);
-        std::vector<unsigned int> evaluateMaxDecisions(unsigned int board);
-        std::vector<unsigned int> evaluateMinDecisions(unsigned int board);
-        static int getScore(const unsigned int &state);
-        static int getState(const unsigned int &board);
-    public:
-        unsigned int evaluate(unsigned int board, bool isMaxTurn);
-        std::vector<unsigned int> evaluateAll(unsigned int board, bool isMaxTurn);
-};
+    const int DRAW = 0;
+    const int CROSS_WINS = 1;
+    const int CIRCLE_WINS = -1;
+    const int PLAYING = 2;
+
+    int predictMaxTreeScore(int board, int alpha, int beta);
+    int predictMinTreeScore(int board, int alpha, int beta);
+    int predictMaxDecision(int board);
+    int predictMinDecision(int board);
+    std::vector<int> predictMaxDecisions(int board);
+    std::vector<int> predictMinDecisions(int board);
+    int getState(int board);
+    int predict(int board, bool isMaxTurn);
+    std::vector<int> evaluateAll(int board, bool isMaxTurn);
```

Comment:
Call it `predictAll` and move `predict` and `predictAll` to the top of the list.